abubakarsadeekimamzy/SportPredix
# Simplify Sports Betting Prediction Market Contract

## Summary of Changes
- Consolidated all functionality into a single Clarity contract (`sports-betting.clar`)
- Removed separate oracle management and token contracts
- Updated the React component (`SportsBettingMarket.tsx`) to work with the simplified contract
- Updated and simplified the test file (`sports-betting.test.ts`)

## Rationale
The previous implementation with separate contracts for oracle management and token functionality introduced unnecessary complexity and potential integration issues. By consolidating all functionality into a single contract, we've achieved:

1. Simplified contract interactions
2. Reduced deployment complexity
3. Improved maintainability
4. Direct use of STX for betting, eliminating the need for a separate token

This approach aligns better with the current scope of the project and allows for easier future expansions if needed.

## Changes in Detail

### Contract (`sports-betting.clar`)
- Integrated oracle management directly into the main contract
- Replaced custom token with direct STX transfers
- Simplified market creation, betting, and settlement processes
- Added functions for oracle registration and removal

### React Component (`SportsBettingMarket.tsx`)
- Updated function calls to match the new contract structure
- Removed token-specific interactions
- Simplified the UI to reflect the consolidated contract functionality

### Tests (`sports-betting.test.ts`)
- Updated test cases to align with the new contract structure
- Added tests for oracle management functions
- Simplified mocking and test setup

## Testing Instructions
1. Deploy the `sports-betting.clar` contract to a Stacks testnet
2. Update the `contractAddress` in `SportsBettingMarket.tsx` with the deployed contract address
3. Run the React application and test the following flows:
   - Create a new market
   - Place bets on the market
   - Register an oracle
   - Settle a market using the registered oracle
   - Claim winnings for a winning bet
4. Run the test suite using `npm test` or `yarn test` to ensure all tests pass

## Potential Impacts and Considerations
- Existing markets and bets from the previous version will not be compatible with this update
- The simplified contract may have different gas costs and execution limits compared to the previous multi-contract setup
- Future expansions, such as more complex oracle systems or custom tokens, may require significant refactoring

## Next Steps
- Implement a more robust oracle system with multiple data sources
- Add support for multiple betting options beyond binary choices
- Enhance the UI with real-time updates and more detailed market information
- Implement additional security measures and access controls

Please review these changes and provide any feedback or suggestions for further improvements.
